_posts/2022/2022-12-05-find-list-debug-active-record-callbacks-in-the-console.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Find out what callbacks are defined on an Active Record model in the console"
+description: "Sometimes it's hard to know what is defined where"
+layout: article
+category: ruby
+image:
+  base: "2022/find-list-debug-active-record-callbacks-in-the-console"
+  alt: "Three old payphones"
+  credit: "Pavan Trikutam"
+  source: "https://unsplash.com/photos/71CjSSB83Wo"
+---
+
+I was recently asked by my [First #RubyFriend](https://firstrubyfriend.org) mentee how to list the callbacks present on an Active Record model. I didn't know.
+
+They were looking at a vast legacy codebase with some large models defined across multiple files. The models used various gems, custom callbacks and concerns. It was very hard to parse what methods and behaviours the callbacks were causing.
```

Comment:
Oxford comma?? 🥺

Reply:
"...what methods" sounds a little awkward alongside "causing". Maybe just stick to "behaviours"?

Reply:
```suggestion
They were looking at a vast legacy codebase, with some large models defined across multiple files. The models used various gems, custom callbacks, and concerns. It was very hard to parse what behaviours the callbacks were causing.
```